src/main/java/org/shanerx/tradeshop/listeners/ShopCreateListener.java

```diff
@@ -79,9 +78,7 @@ public void onSignChange(SignChangeEvent event) {
 			return;
 		}
 
-		JsonConfiguration chunk = new JsonConfiguration(shopSign.getChunk());
-
-		if (Setting.MAX_SHOPS_PER_CHUNK.getInt() <= chunk.getShopCount() + 1) {
+		if (Setting.MAX_SHOPS_PER_CHUNK.getInt() <= plugin.getDataStorage().getShopCountInChunk(shopSign.getChunk()) + 1) {
```

Comment:
what's this changing?

Reply:
it changes to use the DataStorage object rather then having the method in ShopChunk call from JsonConfiguration